src/client/GameSave.cpp

```diff
@@ -1350,18 +1315,10 @@ void GameSave::readPSv(const std::vector<char> &dataVec)
 		}
 	}
 
-	bw = saveData[6];
-	bh = saveData[7];
-	if (bx0+bw > XCELLS)
-		bx0 = XCELLS - bw;
-	if (by0+bh > YCELLS)
-		by0 = YCELLS - bh;
-	if (bx0 < 0)
-		bx0 = 0;
-	if (by0 < 0)
-		by0 = 0;
+	auto blockS = Vec2{ int(saveData[6]), int(saveData[7]) };
+	blockP = blockP.Clamp(blockS.OriginRect());
 
-	if (saveData[5]!=CELL || bx0+bw>XCELLS || by0+bh>YCELLS)
+	if (saveData[5]!=CELL || blockP.X+blockS.X>XCELLS || blockP.Y+blockS.Y>YCELLS)
```

Comment:
```suggestion
	if (saveData[5]!=CELL || !CELLS.OriginRect().Contains(blockP + blockS))
```

Reply:
That's off by one; it implements `>=`.

Reply:
You right. I was thinking of Rect::Contains(RectSized(blockP, blockS)) which would've been right here.